src/itn_uptime_analyzer/identity.go

```diff
@@ -1,141 +1,144 @@
-		}
+        for _, obj := range page.Contents {
+            submissionTime, err := GetSubmissionTime(*obj.Key)
+            if err != nil {
+                log.Fatalf("Error parsing time: %v\n", err)
+            }
 
-		for _, obj := range page.Contents {
-			submissionTime, err := time.Parse(time.RFC3339, (*obj.Key)[32:52]) // Between characters 32 and 52 lies the timestamp of a submission
-			if err != nil {
-				log.Fatalf("Error parsing time: %v\n", err)
-			}
+            if (submissionTime.After(config.Period.Start)) && (submissionTime.Before(config.Period.End)) {
 
-			if (submissionTime.After(lastExecutionTime)) && (submissionTime.Before(currentTime)) {
+                var identity Identity
 
-				var identity Identity
+                objHandle, err := ctx.Client.GetObject(ctx.Context, &s3.GetObjectInput{
+                    Bucket: ctx.BucketName,
+                    Key:    obj.Key,
+                })
 
-				objHandle, err := ctx.Client.GetObject(ctx.Context, &s3.GetObjectInput{
-					Bucket: ctx.BucketName,
-					Key:    obj.Key,
-				})
+                if err != nil {
+                    log.Fatalf("Error getting object from bucket: %v\n", err)
+                }
 
-				if err != nil {
-					log.Fatalf("Error getting object from bucket: %v\n", err)
-				}
+                defer objHandle.Body.Close()
 
-				defer objHandle.Body.Close()
+                objContents, err := io.ReadAll(objHandle.Body)
+                if err != nil {
+                    log.Fatalf("Error getting creating reader for json: %v\n", err)
+                }
 
-				objContents, err := io.ReadAll(objHandle.Body)
-				if err != nil {
-					log.Fatalf("Error getting creating reader for json: %v\n", err)
-				}
+                err = json.Unmarshal(objContents, &submissionData)
+                if err != nil {
+                    log.Fatalf("Error unmarshaling bucket content: %v\n", err)
+                }
 
-				err = json.Unmarshal(objContents, &submissionData)
-				if err != nil {
-					log.Fatalf("Error unmarshaling bucket content: %v\n", err)
-				}
+                var remoteAddr string
+                if config.IgnoreIPs {
+                    remoteAddr = ""
+                } else {
+                    remoteAddr = submissionData.RemoteAddr
+                }
 
-				if submissionData.GraphqlControlPort != 0 {
-					identity = GetFullIdentity(submissionData.Submitter.String(), submissionData.RemoteAddr, strconv.Itoa(submissionData.GraphqlControlPort))
-				} else {
-					identity = GetPartialIdentity(submissionData.Submitter.String(), submissionData.RemoteAddr)
-				}
+                if (!config.IgnoreIPs || submissionData.GraphqlControlPort != 0) {
+                    identity = GetFullIdentity(submissionData.Submitter.String(), remoteAddr, strconv.Itoa(submissionData.GraphqlControlPort))
+                } else {
+                    identity = GetPartialIdentity(submissionData.Submitter.String(), remoteAddr)
```

Comment:
`publicIP` is required in that data structure. We would have to change it to a pointer, so that we can pass `nil`. I don't think it's needed, though. We don't display the IP if `IgnoreIPs` is set, so it only matters that all the IPs are the same so that they compare as equal. I'm against complicating the script more than necessary.